### ISSUE TITLE
More input to use --endpoint-id with command "az monitor data-collection rule association create"

### ISSUE THREAD

**Issue by unixsubhasish17** [Azure/azure-cli#28338]
As per the page, options are given to use --data-collection-rule-id & --data-collection-endpoint-id but its no where mentioned that both options can't be use together.
#az monitor data-collection rule association create --name "NAME_OF_DCR_RULE" --rule-id "DCR_RULE_ID" --resource "VM_RESOURCE_ID" with this command, you cannot use  --data-collection-endpoint-id "endpoint_resource_id" . Error will be **"Association name for resource to endpoint must be configurationAccessEndpoint"**

Now, to resolve this error, you need to associate DCR and DCE separately. You cannot use it together.
So, to Associate VM with DCR,
#az monitor data-collection rule association create --name "NAME_OF_DCR_RULE" --rule-id "DCR_RULE_ID" --resource "VM_RESOURCE_ID"

Now, time to associate endpoint with the VM,
#az monitor data-collection rule association create --name "configurationAccessEndpoint" --resource "VM_RESOURCE_ID" --endpoint-id "endpoint_ID"

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by AllyW** [comment]
@unixsubhasish17  yes, that's restricted by service side even if cli removes the name validation part. 

Azure service api would raise this "InvalidEndpointAssociationName" error when endpoint-id is provided in request body but association name is anything but not "configurationAccessEndpoint".  If it bothers you too much, we can try to raise a feature request to data-collection service team. ;)

**Comment by unixsubhasish17** [comment]
Hello Ally,

Is this possible to raise it and update the documentation?

Thanks
Subhasish
